polkadot/node/core/pvf/src/testing.rs

```diff
@@ -62,6 +62,39 @@ pub fn validate_candidate(
 pub fn get_and_check_worker_paths() -> (PathBuf, PathBuf) {
 	// Only needs to be called once for the current process.
 	static WORKER_PATHS: OnceLock<Mutex<(PathBuf, PathBuf)>> = OnceLock::new();
+
+	fn maybe_build_workers(build_prep: bool, build_exec: bool) {
+		let build_args = match (build_prep, build_exec) {
+			(true, true) => "build --bin polkadot-prepare-worker --bin polkadot-execute-worker",
+			(true, false) => "build --bin polkadot-prepare-worker",
+			(false, true) => "build --bin polkadot-execute-worker",
+			(false, false) => "",
+		};
+
+		if build_args.len() > 0 {
+			eprintln!("Building workers...");
+
+			let mut wd = std::env::current_exe().unwrap();
+			while !wd.ends_with("polkadot-sdk") {
+				wd.pop();
+			}
+
+			let child = match std::process::Command::new("cargo")
+				.args(build_args.split(' '))
+				.stdout(std::process::Stdio::piped())
+				.current_dir(wd)
```

Comment:
```suggestion
				.current_dir(env!("CARGO_MANIFEST_DIR"))
```

If someone points the target dir to somewhere outside of the repo.